src/test/java/seedu/address/storage/StorageManagerTest.java

```diff
@@ -58,6 +62,21 @@ public void prefsReadSave() throws Exception {
         assertEquals(original, retrieved);
     }
 
+    @Test
+    public void handleAddressBookAndEventListChangedEvent_exceptionThrown_eventRaised() {
+        // Create a StorageManager while injecting a stub that  throws an exception when the save method is called
+        Storage storage = new StorageManager(new XmlAddressBookStorageExceptionThrowingStub(Paths.get("dummy")),
+                new XmlEventStorageExceptionThrowingStub(Paths.get("dummy")),
+                new JsonUserPrefsStorage(Paths.get("dummy")));
+        storage.handleAddressBookChangedEvent(new AddressBookChangedEvent(new AddressBook()));
+        assertTrue(eventsCollectorRule.eventsCollector.getMostRecent() instanceof DataSavingExceptionEvent);
```

Comment:
would it be better to separate this into two different tests? for addressbook and eventlist separately